### DISCUSSION TITLE
Video

### DISCUSSION THREAD

**Discussion by sjacksonhodum** [ginbot86/ColorLCDVape-RE#2] [Ideas]
Would It be possible to upload a short video/animation to the screen? If so how?

**Comment by ginbot86** [comment]
In what context? Like replacing the built-in animations, or just displaying a video?

Custom animations can be uploaded with tools made by xbenkozx, and a custom built USB-C cable that connects to a SWD debugging dongle: https://github.com/xbenkozx/RAZ-RE

**Comment by sjacksonhodum** [comment]
I was thinking of running a video on the screen with the built in electronics. Is it possible to recode it to do so

**Reply by ginbot86** [reply]
Theoretically, yes. The 1MB of SPI Flash would only hold about forty 16-bit fullscreen frames, though. Custom firmware would need to be written to do so, mainly dealing with hardware SPI and DMA operations, perhaps even image decompression that's within the limitations of the microcontroller's CPU and RAM limitations.

**Reply by sjacksonhodum** [reply]
Can I take a memory Chip off of a esp32 and use it to get me more space

**Reply by ginbot86** [reply]
Yes, but if it's not a (T)SSOP package then you'll need to use some creativity in wiring up the chip to the existing SPI Flash footprint.

**Reply by sjacksonhodum** [reply]
Thanks for replying. How would I go about getting the video onto the screen?

**Reply by ginbot86** [reply]
Unfortunately this would be something considered as "an exercise for the reader". In a nutshell you'd need a tool to convert your input video to a format the microcontroller can understand and is small enough to fit in SPI Flash, and the microcontroller would need to implement the necessary SPI I/O, data decompression/parsing and DMA functions to display video.
